src/Microsoft.ML.Core/Utilities/ResourceManagerUtils.cs

```diff
@@ -110,7 +110,28 @@ public async Task<ResourceDownloadResults> EnsureResource(IHostEnvironment env,
                     $"Could not create a valid URI from the base URI '{MlNetResourcesUrl}' and the relative URI '{relativeUrl}'");
             }
             return new ResourceDownloadResults(filePath,
-                await DownloadFromUrl(env, ch, absoluteUrl.AbsoluteUri, fileName, timeout, filePath), absoluteUrl.AbsoluteUri);
+                await DownloadFromUrlWithRetry(env, ch, absoluteUrl.AbsoluteUri, fileName, timeout, filePath), absoluteUrl.AbsoluteUri);
+        }
+
+        private async Task<string> DownloadFromUrlWithRetry(IHostEnvironment env, IChannel ch, string url, string fileName,
+            int timeout, string filePath, int retryTimes = 5)
+        {
+            var downloadResult = "";
+
+            for (int i = 0; i < retryTimes; ++i)
+            {
+                var thisDownloadResult = DownloadFromUrl(env, ch, url, fileName, timeout, filePath);
+
+                if (string.IsNullOrEmpty(await thisDownloadResult))
+                    return await thisDownloadResult;
+                else
+                    downloadResult += thisDownloadResult.Result + @"\n";
+
+                Random random = new Random();
+                Thread.Sleep(random.Next(1, 60) * 1000);
```

Comment:
💭 This is an extremely unusual delay strategy. I would expect constant, progressive backoff, or exponential backoff.